source/common/quic/envoy_quic_server_stream.h

```diff
@@ -18,7 +20,11 @@ class EnvoyQuicServerStream : public quic::QuicSpdyServerStreamBase,
                         envoy::config::core::v3::HttpProtocolOptions::HeadersWithUnderscoresAction
                             headers_with_underscores_action);
 
-  void setRequestDecoder(Http::RequestDecoder& decoder) { request_decoder_ = &decoder; }
+  void setRequestDecoder(Http::RequestDecoder& decoder) override {
+    request_decoder_ = &decoder;
+    stats_gatherer_->setStreamInfo(request_decoder_->streamInfoSharedPtr());
```

Comment:
As we update the request_decoder_ reference in the stream now, can the QuicStatsGatherer hold a handle to the stream and get the stream info and access log handlers from the stream in maybeDoDeferredLog() instead?

Reply:
To answer myself, request_decoder_ short lives the QuicStream so this is not a good way.